### PR TITLE
[Fix_3383] Adding binary cloud event support for outgoing messages

### PR DESCRIPTION
**Pull request by fjtirado**
Knative has issues with structure event, so we are switching to binary, by defaullt, for quarkus-http connector. 

This behaviour can be modified through configuration. 
Users can specifiy
`kogito.addon.messaging.outgoing.cloudEventMode.<channelName>=BINARY|STRUCTURED`
to configure cloud event mode per outgoing channel 
or just 
`kogito.addon.messaging.outgoing.cloudEventMode=BINARY|STRUCTURED`
to set it overrall. 
If nothing specified, binary is used for quarkus-http connector and structured for the others, to keep backward compatibility with existing examples using Kafka. 

This requires changes on kogito-apps tests https://github.com/apache/incubator-kie-kogito-apps/pull/1974 and examples https://github.com/apache/incubator-kie-kogito-examples/pull/1862

Fixes https://github.com/apache/incubator-kie-kogito-runtimes/issues/3383

### CODE REVIEW

File: kogito-codegen-modules/kogito-codegen-core/src/main/resources/class-templates/config/GlobalObjectMapperQuarkusTemplate.java [hunk 1]
```diff
@@ -44,5 +45,6 @@ public void customize(ObjectMapper mapper) {
         }
         mapper.setDateFormat(new StdDateFormat().withColonInTimeZone(true).withTimeZone(TimeZone.getDefault()));
         mapper.registerModule(new JavaTimeModule()).registerModule(JsonFormat.getCloudEventJacksonModule());
+        mapper.setSerializationInclusion(Include.NON_NULL);
```
Comment (ricardozanini) [inline kogito-codegen-modules/kogito-codegen-core/src/main/resources/class-templates/config/GlobalObjectMapperQuarkusTemplate.java:48]: Ins't too risk to add this option globally?

Comment (fjtirado) [reply kogito-codegen-modules/kogito-codegen-core/src/main/resources/class-templates/config/GlobalObjectMapperQuarkusTemplate.java:48]: yes, and it wont fix anything actually. As you know we are actively discussing this issue on slack. Im moving to draft

**Review by ricardozanini — APPROVED** [review]
The code looks fine, I'm just wondering about what Matej said that CE SDK can handle the serialization for specific connectors, like HTTP and Kafka.

**Comment by fjtirado** [comment]
> The code looks fine, I'm just wondering about what Matej said that CE SDK can handle the serialization for specific connectors, like HTTP and Kafka.

@ricardozanini In kogito, it was decided at the beginning to handle the serialization through custom marshallers rather than relying on connector specifics. However, if user want do to that, one of the provided marshallers is a not marshaller at all, which relies on the connector setup

**Pull request merged**